### ISSUE TITLE
Enforce standard formatting

### ISSUE THREAD

**Issue by rparini** [abelfunctions/abelfunctions#206]
For general readability I'd suggest enforcing some minimal static analysis tests for formatting. 

What I've done in other projects, and I think works well, is:
- run black in CI and fail if it would make any changes
  - suggest contributors configure [black](https://github.com/psf/black) to run on save in vscode (or the editor/IDE of their choice) 
  - (optionally) configure black to run in a git pre-commit hook (this will avoid black failures in ci but then a contributor *has* to have black installed to commit anything)
- run [flake8](https://github.com/pycqa/flake8) in CI and fail on errors (we can confgure `pyproject.toml` to ignore some classes of errors or ignore individual errors in the code)

I think [Black won't work on `.pyx` files](https://github.com/psf/black/issues/359) and flake8 might need some adjusting. We could ignore those files in the first go at this if it's hard.

**Comment by cswiercz** [comment]
I like the idea of normalizing the format. I haven't used either black or flake8 so I have no preference. Some requirements/tasks for us for this ticket:

[ ] (P0) Include instructions or at least list the formatter requirement somewhere in documentation / readthedocs / developer guide (e.g. how to setup black to run in vscode)
[ ] (P1) Set up a git pre-commit hook or make it part of the commit test suite (e.g. build fails if formatted code doesn't match committed code)
[ ] (P2) Create a custom `build` script that will automatically run black before compiling and/or running the test suite (a local hook)

**Comment by rparini** [comment]
>  [ ] (P2) Create a custom build script that will automatically run black before compiling and/or running the test suite (a local hook)

What use case did you have in mind for this? Black isn't going to make any functional changes, it's just formatting, so I don't think I would be bothered about ensuring its run before compiling & running tests locally. I think it just needs to be run before pushing (either manually with `black .` or with a pre-commit hook).

**Comment by cswiercz** [comment]
Use case is just automating the process. Easy to forget as a developer and easy to be unaware of as a non-developer. I worked with a repo that had pre-commit hooks and it was frustrating when you forgot. However, the build script would either auto-format the code when run or at least give you a local error that the formatter hadn't been run.

This was marginally more convenient than pushing and having CI complain. I'm fine dropping this requirement. (I listed as P2.)
